### PR TITLE
[New Model] Add Seed-Oss model

### PR DESCRIPTION
**Pull request by FoolPlayer**
## Purpose
Add Seed-Oss model.
transformers see: https://github.com/huggingface/transformers/pull/40272

## (Optional) Documentation Update

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by youkaichao** [comment]
cc @DarkLight1337 who can help take a look?

File: vllm/model_executor/models/seed_oss.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+                    continue
+                param = params_dict[name]
+                weight_loader = getattr(param, "weight_loader",
+                                        default_weight_loader)
+                weight_loader(param, loaded_weight)
+            loaded_params.add(name)
+        return loaded_params
+
+
+class SeedOssForCausalLM(nn.Module, SupportsLoRA, SupportsPP):
```
Comment (jeejeelee) [inline vllm/model_executor/models/seed_oss.py:414]: It looks like this model structure is consistent with llama. Can we directly inherit from llama, just like [phi3](https://github.com/vllm-project/vllm/blob/v0.10.1/vllm/model_executor/models/phi3.py#L10) does?

File: vllm/entrypoints/openai/tool_parsers/seed_oss_tool_parser.py [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+        tool_start_re = re.escape(self.tool_call_start_token)
+        tool_end_re = re.escape(self.tool_call_end_token)
+
+        self.tool_call_complete_regex = re.compile(
+            rf"{tool_start_re}(.*?){tool_end_re}", re.DOTALL)
+        self.tool_call_regex = re.compile(
+            rf"{tool_start_re}(.*?){tool_end_re}|{tool_start_re}(.*?)$",
+            re.DOTALL)
+
+        # 其余 regex 与原版一致
```
Comment (jeejeelee) [inline vllm/entrypoints/openai/tool_parsers/seed_oss_tool_parser.py]: We should add comment in english

File: vllm/model_executor/models/seed_oss.py (hunk 1 above)
Comment (jeejeelee) [reply vllm/model_executor/models/seed_oss.py:414]: We also need to update the [supported_models](https://github.com/vllm-project/vllm/blob/main/docs/models/supported_models.md#list-of-text-only-language-models) and [test](https://github.com/vllm-project/vllm/blob/main/tests/models/registry.py#L138)

File: vllm/entrypoints/openai/tool_parsers/seed_oss_tool_parser.py [hunk 3]
```diff
@@ -0,0 +1,632 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (jeejeelee) [inline vllm/entrypoints/openai/tool_parsers/seed_oss_tool_parser.py:1]: We need to  add test for this tool parser, see: https://github.com/vllm-project/vllm/tree/main/tests/tool_use

File: vllm/model_executor/model_loader/utils.py [hunk 4]
```diff
@@ -189,6 +189,8 @@ def get_model_architecture(
         architectures,
         model_config=model_config,
     )
+    print(architectures)
+    print(model_config)
```
[the remark this answers is not in the document]
Comment (jeejeelee) [reply vllm/model_executor/model_loader/utils.py]: Makes sense

**Comment by ZJY0516** [comment]
Now that Seed-OSS is released, I was wondering if there's any progress on this PR.

**Comment by FoolPlayer** [comment]
> Now that Seed-OSS is released, I was wondering if there's any progress on this PR.

thanks for review, i'll fix the comments these days

**Review by jeejeelee — APPROVED** [review]
Already confirmed with @FoolPlayer, both the generation and tool calls are working fine, so I'm adding my stamp.
@FoolPlayer  Thank you for contribuion

**Comment by prnake** [comment]
In addition to the documents and test files, the current PR is lacking some necessary elements, such as ReasoningParser.

The current PR does not handle the xml tags such as <seed:think>.

```
{"id":"chatcmpl-7f45361eec5e483c96a119683efdadd7","object":"chat.completion","created":1755792348,"model":"Seed-OSS-36B-Instruct","choices":[{"index":0,"message":{"role":"assistant","content":"<seed:think>\nGot it, let's see. The user wrote \"Ciallo\". First, I need to figure out what that means. \"Ciallo\" is a term I've heard before, maybe from Japanese or anime? Wait, let me think. Oh right, \"ciao\" is Italian for \"hello\" or \"goodbye\", but \"ciallo\"—maybe it's a misspelling or a different language. Wait, no, actually, in some anime or Japanese contexts, \"ciallo\" is sometimes used as a casual way to say \"hello\" or \"hi\", maybe influenced by \"ciao\"? Or is it a typo? Wait, let me check. Alternatively, maybe it's a name? Or a slang term. Wait, another angle: \"ciallo\" could be a transliteration of a Japanese sound or a term. Let me think... Oh, right! There's a character named Ciallo? Or maybe it's a fanservice term. Wait, no, actually, I remember that \"ciallo\" is sometimes used in online communities, especially among anime fans, as a friendly greeting, similar to \"hi\" or \"hello\". It might have originated from a mishearing or a playful twist on \"ciao\". Alternatively, could it be a typo for \"ciao\"? But the user wrote \"ciallo\" with two Ls. Let me confirm. Looking it up quickly in my mind: yes, \"ciallo\" is often used in casual, friendly contexts, especially in anime or otaku communities, as a way to say hello. So the user probably just said \"hello\" in a casual, maybe anime-influenced way. So the response should be friendly, acknowledging that. Maybe say \"Hi there! Ciallo is a cute greeting, right? How are you doing today?\" or something like that. Keep it warm and welcoming.</seed:think>Ciallo! It’s such a cheerful, friendly little greeting—often seen in casual anime or otaku circles, like a playful \"hi\" or \"hello.\"  \n\nHow are you feeling today? 😊","refusal":null,"annotations":null,"audio":null,"function_call":null,"tool_calls":[],"reasoning_content":null},"logprobs":null,"finish_reason":"stop","stop_reason":null}],"service_tier":null,"system_fingerprint":null,"usage":{"prompt_tokens":10,"total_tokens":441,"completion_tokens":431,"prompt_tokens_details":null},"prompt_logprobs":null,"kv_transfer_params":null}
```
